bootstrap-salt.sh

```diff
@@ -4325,6 +4353,179 @@ install_freebsd_restart_daemons() {
+          # [ $fname = "api" ] && ([ "$_INSTALL_MASTER" -eq $BS_FALSE ] || [ "$(which salt-${fname} 2>/dev/null)" = "" ]) && continue
+          [ $fname = "syndic" ] && [ "$_INSTALL_SYNDIC" -eq $BS_FALSE ] && continue
+
+          # Let's download, since they were not provided, the default configuration files
+          if [ ! -f "$_SALT_ETC_DIR/$fname" ] && [ ! -f "$_TEMP_CONFIG_DIR/$fname" ]; then
+              ftp -o "$_TEMP_CONFIG_DIR/$fname" \
+                  "https://raw.githubusercontent.com/saltstack/salt/develop/conf/$fname" || return 1
+          fi
+      done
+  fi
+  if [ "$_UPGRADE_SYS" -eq $BS_TRUE ]; then
+      /usr/local/bin/pip install --upgrade pip || return 1
+  fi
+  if [ "${_EXTRA_PACKAGES}" != "" ]; then
+      echoinfo "Installing the following extra packages as requested: ${_EXTRA_PACKAGES}"
+      # shellcheck disable=SC2086
+      pkg_add -I -v ${_EXTRA_PACKAGES} || return 1
+  fi
+  return 0
+}
+
+install_openbsd_git_deps() {
+    install_openbsd_deps || return 1
+    pkg_add -I -v git || return 1
+    __git_clone_and_checkout || return 1
+    #
+    # Let's trigger config_salt()
+    #
+    if [ "$_TEMP_CONFIG_DIR" -eq "null" ]; then
+        _TEMP_CONFIG_DIR="${__SALT_GIT_CHECKOUT_DIR}/conf/"
+        CONFIG_SALT_FUNC="config_salt"
+    fi
+    return 0
+}
+
+install_openbsd_stable() {
+    
+#    pkg_add -r -I -v salt || return 1 
+    check_pip_allowed
+      /usr/local/bin/pip install salt || return 1
+     if [ "$_UPGRADE_SYS" -eq $BS_TRUE ]; then
+         /usr/local/bin/pip install --upgrade salt || return 1
+     fi
+    return 0
+}
+
+install_openbsd_git() {
+    #
+    # Install from git
+    #
+    if [ ! -f salt/syspaths.py ]; then
+        # We still can't provide the system paths, salt 0.16.x
+        /usr/local/bin/python2.7 setup.py install || return 1
+    fi
+    return 0
+}
+
+
+install_openbsd_post() {
+
+    #
+    # Install rc.d files.
+    #
+    ## below workaround for /etc/rc.d/rc.subr in OpenBSD >= 5.8 
+    ## is needed for salt service to start/stop properly by /etc/rc.d
+    ## reference: http://cvsweb.openbsd.org/cgi-bin/cvsweb/src/etc/rc.d/rc.subr.diff?r1=1.98&r2=1.99
+    ##
+    if [ $(grep '\-xf' /etc/rc.d/rc.subr)]; then
+      cp -p /etc/rc.d/rc.subr /etc/rc.d/rc.subr
+      sed -i.$(date +%F).saltinstall -e 's:-xf:-f:g' /etc/rc.d/rc.subr
+    fi
+    _TEMP_CONFIG_DIR="/tmp"
+    for fname in minion master syndic api; do
+
+        # Skip if not meant to be installed
+        [ $fname = "minion" ] && [ "$_INSTALL_MINION" -eq $BS_FALSE ] && continue
+        [ $fname = "master" ] && [ "$_INSTALL_MASTER" -eq $BS_FALSE ] && continue
+        [ $fname = "api" ] || [ "$(which salt-${fname} 2>/dev/null)" = "" ] && continue
+        #[ $fname = "api" ] && ([ "$_INSTALL_MASTER" -eq $BS_FALSE ] || [ "$(which salt-${fname} 2>/dev/null)" = "" ]) && continue
+        #[ $fname = "syndic" ] && [ "$_INSTALL_SYNDIC" -eq $BS_FALSE ] && continue
```

Comment:
@serge-p By the way, we have `__check_command_exists` function in the script. Better to use it instead of `which`, like this:
```
[ $fname = "api" ] || __check_command_exists salt-${fname} && continue
```